logger/Include/Logger.cpp

```diff
@@ -0,0 +1,260 @@
+#include "Macros.h"
+#include "Logger.h"
+
+//std::unique_ptr<Logger> Logger::instance = nullptr;
+
+Logger::Logger(const LogLevel& level, const std::string& path, const unsigned int amount, const bool do_flush)
```

Comment:
Please use std types to express what data type you actually want to use. Again, that depends on compiler and architecture your program is going to run on, but it's a good rule of thumb. That applies to all other uses of primitive types.

```suggestion
Logger::Logger(const LogLevel& level, const std::string& path, const std::uint32_t amount, const bool do_flush)
```